src/Razor/src/Microsoft.AspNetCore.Razor.Language/DefaultRazorProjectFileSystem.cs

```diff
@@ -68,9 +68,11 @@ protected override string NormalizeAndEnsureValidPath(string path)
                 throw new ArgumentException(Resources.ArgumentCannotBeNullOrEmpty, nameof(path));
             }
 
-            var absolutePath = path;
-            if (!absolutePath.StartsWith(Root, StringComparison.OrdinalIgnoreCase))
+            var absolutePath = path.Replace('\\', '/');
+            if (!absolutePath.StartsWith(Root, StringComparison.OrdinalIgnoreCase) &&
+                !absolutePath.StartsWith("//", StringComparison.OrdinalIgnoreCase))
```

Comment:
This is needed because of cases like https://github.com/aspnet/AspNetCore-Tooling/pull/298/files#diff-3981779c97d15dc208b6d34604acb7b2R272